src/Integration.Vsix.UnitTests/CFamily/CLangAnalyzerTests.cs

```diff
@@ -100,6 +104,99 @@ public void ExecuteAnalysis_ValidProjectItem_RequestCanBeCreated_AnalysisIsTrigg
             testSubject.TriggerAnalysisCallCount.Should().Be(1);
         }
 
+        [TestMethod]
+        public void TriggerAnalysisAsync_StreamsIssuesFromSubProcessToConsumer()
```

Comment:
The new tests have to call the analyzer on a different thread from the test thread, and then selectively block/wait as the messages are processed.
As far as I can tell, the tests are reliable (i.e. repeatably run correctly without flakiness) and also clean up properly in the event of errors.